src/foraging_gui/MyFunctions.py

```diff
@@ -1166,26 +1166,26 @@ def _CheckStop(self):
         stop = False
         msg =''
         warning_label_text = ''
-        warning_label_color = 'color: gray;'        
 
         # Check for reasons to stop early 
-        if (np.shape(self.B_AnimalResponseHistory)[0]>=StopIgnore) and (np.all(self.B_AnimalResponseHistory[-StopIgnore:]==2)):
+        non_auto_reward = self.B_AnimalResponseHistory[np.where(~self.TP_AutoReward)]   # isolate non-auto-reward trials
+        win_sz = self.TP_auto_stop_ignore_win
+        if self.BS_CurrentRunningTime >= 30 and \
```

Comment:
@hanhou is it okay that the session time check is hard coded to 30? Would there be any reason to make it a training parameter?

Reply:
Making it as a parameter would be great